test/recipes/25-test_x509.t

```diff
@@ -187,18 +189,44 @@ ok(!run(app(["openssl", "x509", "-noout", "-dates", "-dateopt", "invalid_format"
 	     "-in", srctop_file("test/certs", "ca-cert.pem")])),
    "Run with invalid -dateopt format");
 
-# extracts issuer from a -text formatted-output
-sub get_issuer {
-    my $f = shift(@_);
-    my $issuer = "";
+# extracts string value of certificate field from a -text formatted-output
+sub get_field {
+    my ($f, $field) = @_;
+    my $string = "";
     open my $fh, $f or die;
     while (my $line = <$fh>) {
-        if ($line =~ /Issuer:/) {
-            $issuer = $line;
+        if ($line =~ /$field:\s+(.*)/) {
+            $string = $1;
         }
     }
     close $fh;
-    return $issuer;
+    return $string;
+}
+
+sub get_issuer {
+    return get_field(@_, "Issuer");
+}
+
+sub get_not_before {
+    return get_field(@_, "Not Before");
+}
+
+# Date as yyyy-mm-dd
+sub get_not_before_date {
+    return Time::Piece->strptime(
+	get_not_before(@_),
+	"%b %d %T %Y %Z")->date;
+}
+
+sub get_not_after {
+    return get_field(@_, "Not After ");
+}
+
+# Date as yyyy-mm-dd
+sub get_not_after_date {
+    return Time::Piece->strptime(
+	get_not_after(@_),
+	"%b %d %T %Y %Z")->date;
```

Comment:
Very nice - this provides a more general alternative to using  the `cert_contains()` subroutine.
Please move these new functions to `tconversion.pl` such that they are usable also for `req` (and possibly further test in the future).

Reply:
Moved to `tconversion.pl`.

Reply:
Hmm, I still see them here in `25-test_x509.t`.

Reply:
Sorry, my fault! I also moved the remaining `get_issuer()` function now.